service/frontend/workflowHandler.go

```diff
@@ -1188,8 +1193,7 @@ func (wh *WorkflowHandler) RecordActivityTaskHeartbeat(ctx context.Context, requ
 		// heartbeat details exceed size limit, we would fail the activity immediately with explicit error reason
 		failRequest := &workflowservice.RespondActivityTaskFailedRequest{
 			TaskToken: request.TaskToken,
-			Reason:    common.FailureReasonHeartbeatExceedsLimit,
-			Details:   nil,
+			Failure:   failure.NewServerFailure(common.FailureReasonHeartbeatExceedsLimit, true),
```

Comment:
Setting isRetryable to `true` for size exceeded could be really dangerous.

Reply:
The field is actually called `nonRetryable`, and `true` means "don't retry". Logic is stayed the same here.